src/main/scala/io/lenses/connect/secrets/providers/VaultSecretProvider.scala

```diff
@@ -8,33 +8,44 @@ package io.lenses.connect.secrets.providers
 
 import java.nio.file.FileSystems
 import java.time.OffsetDateTime
-import java.{io, lang, util}
+import java.util
 
-import com.bettercloud.vault.Vault
-import _root_.io.lenses.connect.secrets.config.{
-  VaultProviderConfig,
-  VaultSettings
-}
+import _root_.io.lenses.connect.secrets.config.VaultProviderConfig
+import _root_.io.lenses.connect.secrets.config.VaultSettings
 import _root_.io.lenses.connect.secrets.connect._
+import com.bettercloud.vault.Vault
+import io.lenses.connect.secrets.async.AsyncFunctionLoop
 import org.apache.kafka.common.config.ConfigData
 import org.apache.kafka.common.config.provider.ConfigProvider
 import org.apache.kafka.connect.errors.ConnectException
 
 import scala.collection.JavaConverters._
 import scala.collection.mutable
-import scala.util.{Failure, Success, Try}
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
 
 class VaultSecretProvider() extends ConfigProvider with VaultHelper {
 
   private val separator: String = FileSystems.getDefault.getSeparator
   private var settings: VaultSettings = _
-  var vaultClient: Option[Vault] = None
-  val cache = mutable.Map.empty[String, (Option[OffsetDateTime], ConfigData)]
+  private var vaultClient: Option[Vault] = None
+  private var tokenRenewal: Option[AsyncFunctionLoop] = None
+  private val cache = mutable.Map.empty[String, (Option[OffsetDateTime], ConfigData)]
+
+  def getClient: Option[Vault] = vaultClient
 
   // configure the vault client
   override def configure(configs: util.Map[String, _]): Unit = {
     settings = VaultSettings(VaultProviderConfig(configs))
     vaultClient = Some(createClient(settings))
+    tokenRenewal = Some(new AsyncFunctionLoop(settings.tokenRenewal, "Vault Token Renewal")(renewToken()))
```

Comment:
I'm not sure where `tokenRenewal` get started some where? I pull the code and run the issue of lease duration still occurred